### DISCUSSION TITLE
Quick terminal always contains bottom padding with `window-step-resize = true`

### DISCUSSION THREAD

**Discussion by JakeWharton** [ghostty-org/ghostty#2946] [Feature Requests, Ideas] [answered]
I'm on macOS 14.7.1 (23H222) with Ghostty 9f75d93a.

My full config is
```
initial-window = false
copy-on-select = true
window-save-state = always
background-opacity = 0.9
background-blur-radius = 5
window-step-resize = true
keybind = global:alt+grave_accent=toggle_quick_terminal
mouse-hide-while-typing = true
macos-option-as-alt = true
```

The quick terminal seems to always have bottom padding when snapping to the character grid is enabled.

![Image](https://github.com/user-attachments/assets/b21adcb5-8685-414c-bb2e-de7dbe15c0de)

This persists across resizing:

![Image](https://github.com/user-attachments/assets/1c9236b4-2fa6-4e38-9fa3-276cf1ff75f4)

**Comment by mitchellh** [comment]
#2922 is related. 

I think this may also warrant discussion rather than being actionable yet.

**Reply by stepanbujnak** [reply]
Discussion about whether this is actually a problem or whether setting the default window size is the right approach?

**Reply by mitchellh** [reply]
A broader discussion about how `window-step-resize` behaves in general I think with new windows, animated native widgets that may change the viewport, etc. 

I think the general consensus around people who use this config or are used to things like Terminal.app that force it is that they _expect_ their window to change size in order to maintain the locked grid size.

**Reply by JakeWharton** [reply]
I can only speak for myself, but that is how I interpreted it and is the behavior I was trying to achieve.

I thought that when I resize the terminal viewport to a particular character size, that size would remain fixed as the various window chrome around it comes in and out, or changes in some way (such as tab bar showing or hiding).

**Comment by stepanbujnak** [comment]
Just an extra bit of information about #2919. As I already mentioned, this only happens when we set the `contentResizeIncrements` to a fraction, e.g. `16.5`. I looked at the sizes reported from `updateOSView` and they increment by `16`, basically discarding the `.5`. Looking at the documentation for `contentResizeIncrements` it doesn't mention anything about that, but the documentation for `resizeIncrements` says:

>As the user resizes the window, its size changes by multiples of increments``.width and increments``.height, which should be whole numbers, 1.0 or greater.

Which makes me think that this just doesn't work by design. If I'm correct (very possible that I am not), the only option would be to make sure that the cell size is set to a value that will be divisible by the scale factor. Or we ignore the problem but that makes the resizing kinda wonky.

**Reply by mitchellh** [reply]
Thanks @stepanbujnak, this is an excellent find. I think this warrants its own discussion so we don't lose track of this, although I'm pretty ready to make an issue about it I want to do a bit more research.

**Comment by mitchellh** [comment, marked as answer]
Issue: #5300
